shopinvader_api_payment/routers/payment.py

```diff
@@ -0,0 +1,197 @@
+# Copyright Odoo SA (https://odoo.com)
+# Copyright 2024 ACSONE SA (https://acsone.eu).
+# @author Stéphane Bidoul <stephane.bidoul@acsone.eu>
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+from typing import Annotated
+from urllib.parse import urljoin
+
+from fastapi import APIRouter, Depends, HTTPException, Request
+
+from odoo import api, models
+
+from odoo.addons.fastapi.dependencies import odoo_env
+from odoo.addons.payment import utils as payment_utils
+from odoo.addons.payment.models.payment_provider import PaymentProvider
+
+from ..schemas import (
+    PaymentDataWithProviders,
+    PaymentProvider as PaymentProviderSchema,
+    TransactionCreate,
+    TransactionProcessingValues,
+)
+
+_logger = logging.getLogger(__name__)
+
+payment_router = APIRouter(tags=["payment"])
+
+
+@payment_router.get("/payment/providers")
+def pay(
+    payable: str,  # model,id
+    payable_reference: str,
+    amount: float,
+    currency_id: int,
+    partner_id: int,
+    company_id: int,
+    access_token: str,
+    odoo_env: Annotated[api.Environment, Depends(odoo_env)],
+):
+    """Available payment providers.
+
+    This route is public, so it is possible to pay anonymously provided that the
+    parameters are obtained securely by another mean. An authenticated use can
+    obtain the parameters with the /payment/payable route.
+    """
+    # This method is similar to Odoo's PaymentPortal.payment_pay
+    if partner_id:
+        if not payment_utils.check_access_token(
+            access_token,
+            payable,
+            payable_reference,
+            amount,
+            currency_id,
+            partner_id,
+            company_id,
+        ):
+            _logger.info("Invalid access token for partner %s", partner_id)
+            raise HTTPException(403)
+    providers_sudo = (
+        odoo_env["payment.provider"]
+        .sudo()
+        ._get_compatible_providers(
+            company_id, partner_id, amount, currency_id=currency_id
+        )
+    )
+    return PaymentDataWithProviders(
+        payable=payable,
+        payable_reference=payable_reference,
+        amount=amount,
+        currency_id=currency_id,
+        partner_id=partner_id,
+        company_id=company_id,
+        access_token=access_token,
+        providers=[
+            PaymentProviderSchema.from_payment_provider(provider)
+            for provider in providers_sudo
+        ],
```

Comment:
Here we can keep `providers`, and later add another field for the list of payment tokens.